### PR TITLE
Core/Scripts: Rewrite Boss Kalecgos (Sunwell Plateau)

### PR DESCRIPTION
**Pull request by sirikfoll**
**Changes proposed:**

-  Scripted all spells
-  Corrected behavior of Spectral Blast and Curse of Boundless Agony
-  Removed wrong wipe behavior
-  Fix problem where bosses would be stuck with banish
-  Added Boundaries
-  Corrected texts
-  Removed old hacks
-  Other minor fixes

**Target branch(es):** 3.3.5/master

- [x] 3.3.5

**Tests performed:** (Does it build, tested in-game, etc.) Built and tested, with 2 players

### CODE REVIEW

File: src/server/scripts/EasternKingdoms/SunwellPlateau/boss_kalecgos.cpp [hunk 1]
```diff
@@ -16,523 +15,590 @@
...
+                    Talk(SAY_OUTRO_1);
+                    events.ScheduleEvent(EVENT_OUTRO_3, Seconds(9));
                     break;
-                case DO_BANISH:
-                    isBanished = true;
-                    me->CastSpell(me, SPELL_BANISH, true);
+                case POINT_OUTRO_2:
+                    me->SetVisible(false);
+                    DespawnPortals();
+                    me->KillSelf();
```
Comment (clopezramos) [inline src/server/scripts/EasternKingdoms/SunwellPlateau/boss_kalecgos.cpp]: despawnorunsummon

File: src/server/scripts/EasternKingdoms/SunwellPlateau/boss_kalecgos.cpp [hunk 2]
```diff
@@ -572,239 +624,200 @@ class kalecgos_teleporter : public GameObjectScript
...
-                TeleportAllPlayersBack();
-                if (Creature* Kalecgos = ObjectAccessor::GetCreature(*me, KalecgosGUID))
-                {
-                    ENSURE_AI(boss_kalecgos::boss_kalecgosAI, Kalecgos->AI())->TalkTimer = 1;
-                    ENSURE_AI(boss_kalecgos::boss_kalecgosAI, Kalecgos->AI())->isFriendly = false;
-                }
-                EnterEvadeMode();
-                return;
+                Unit* target = GetHitUnit();
+                target->CastSpell(target, SPELL_SPECTRAL_REALM_TELEPORT, true);
```
Comment (Keader) [inline src/server/scripts/EasternKingdoms/SunwellPlateau/boss_kalecgos.cpp]: missing validator for these spells

**Comment by Keader** [comment]
Click in Green button 👇

**Comment by sirikfoll** [comment]
haha, I would like some users feedback first

**Comment by Aokromes** [comment]
Conflicting files
src/server/scripts/EasternKingdoms/SunwellPlateau/boss_kalecgos.cpp

**Comment by qxpf666** [comment]
Conflicting files
src/server/scripts/EasternKingdoms/SunwellPlateau/boss_kalecgos.cpp

**Comment by Aokromes** [comment]
this is pull request is part of trinitycore since 2 years ago.

sirikfoll merged commit e9cd7af into TrinityCore:3.3.5  on Aug 9, 2017

**Comment by qxpf666** [comment]
can someone update it?thanks

**Comment by Keader** [comment]
Update? wtf?

**Comment by Aokromes** [comment]
there is nothing to update, the code is part of trinitycore since 2 years ago, unless you are trying to use some ancient arcaic version we don't support.

**Pull request merged**